components/platform_crypto/shave_trusted_boot/xmss-library/include/structures.h

```diff
@@ -208,9 +208,31 @@ typedef struct XmssSignatureBlob {
      * The signature data in the format described by RFC 8391, Section 4.1.8.
      * This will always contain one signature in the XmssSignature format.
      */
-    uint8_t data[];
+    uint8_t data[1];
 } XmssSignatureBlob;
 
+/*$
+datatype XmssSignatureBlobT {
+  XmssSigBlobF {size_t data_size, map<u64,u8> data}
+}
+predicate (datatype XmssSignatureBlobT) XmssSignatureBlobP(pointer p)
```

Comment:
@podhrmic This block of three functions contains most of my questions about this code:

1. It's not possible to name this predicate the same as the function. I just named it *p*redicate but perhaps *w*ell-*f*ormed would be better.
2. Same with its return type, we really just want a lifted version of `XmssSignatureBlob` that goes as far as we want it to.
3. This `data_size` function is silly but I need something to let me write that the `data_size` element has not changed.

Reply:
2 and 3 are resolved